src/Microsoft.HttpRepl.Tests/HttpStateTests.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.HttpRepl.FileSystem;
+using Microsoft.HttpRepl.Preferences;
+using Microsoft.HttpRepl.UserProfile;
+using Xunit;
+
+namespace Microsoft.HttpRepl.Tests
+{
+    public class HttpStateTests
+    {
+        [Fact]
+        public void GetPathString_EmptyPathSections_Slash()
+        {
+            string expected = "/";
+            HttpState state = SetupHttpState();
+            state.PathSections.Clear();
+
+            string result = state.GetRelativePathString();
+
+            Assert.Equal(expected, result);
+        }
+
+        [Fact]
+        public void GetPathString_SinglePathSection_CorrectString()
+        {
+            string expected = "/FirstDirectory";
+            HttpState state = SetupHttpState();
+            state.PathSections.Push("FirstDirectory");
+
+            string result = state.GetRelativePathString();
+
+            Assert.Equal(expected, result);
+        }
+
+        [Fact]
+        public void GetPathString_MultiplePathSections_CorrectString()
+        {
+            string expected = "/FirstDirectory/SecondDirectory";
+            HttpState state = SetupHttpState();
+            state.PathSections.Push("FirstDirectory");
+            state.PathSections.Push("SecondDirectory");
+
+            string result = state.GetRelativePathString();
+
+            Assert.Equal(expected, result);
+        }
+
+        private static HttpState SetupHttpState()
+        {
+            IFileSystem fileSystem = new RealFileSystem();
```

Comment:
Shouldn't unit tests generally use a fake filesystem?

Reply:
Whoops. Yeah. There wasn't a fake one in this assembly, so I created a true stub (since it's never actually called in these tests) in afc0086. Tried to follow the naming conventions we talked about yesterday.